src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -499,9 +505,18 @@ private void publishNewAddressReportedEvent(
     newAddress.setId(caseId.toString());
     newAddress.setSurvey("CENSUS");
 
-    // TODO Derivation of addressLevel to be clarified. For now set to value covering majority of
-    // cases
-    newAddress.getAddress().setAddressLevel("U");
+    EstabType aimsEstabType = EstabType.forCode(newAddress.getAddress().getEstabType());
```

Comment:
I think we need a slight tweak here - I know this code reflects the original JIRA but the parallel RH work doing something very similar has led to a subtle change in thinking :
`
    EstabType aimsEstabType = EstabType.forCode(newAddress.getAddress().getEstabType());
    Optional<AddressType> addressTypeMaybe = aimsEstabType.getAddressType();
 
    AddressType addressType = addressTypeMaybe.isPresent() ? addressTypeMaybe.get() : address.getAddressType();

       if (addressType == AddressType.HH || addressType == AddressType.SPG) {
         newAddress.getAddress().setAddressLevel(AddressLevel.U.name());
       } else {
         newAddress.getAddress().setAddressLevel(AddressLevel.E.name());
       }
`

Reply:
Done.